elasticsearch/docker/templates/Dockerfile.j2

```diff
@@ -50,6 +53,42 @@ COPY --chown=1000:0 {{ '%s/' % artifacts_dir }} {{ artifacts_dir }}
 {% endif %}
 
 WORKDIR /usr/share/elasticsearch
+##Compile the C-library for kNN
+RUN set -ex; \
+ \
+    git clone https://github.com/nmslib/nmslib.git /usr/share/elasticsearch/nmslib; \
+ \
+    cd /usr/share/elasticsearch/nmslib \
+ \
+    && git checkout tags/v1.7.3.6 \
+ \
+    && cd similarity_search \
+ \
+    && cmake . \
+ \
+    && make \
+ \
+    && git clone https://github.com/opendistro-for-elasticsearch/k-NN.git /usr/share/elasticsearch/k-NN \
+ \
+    && cd /usr/share/elasticsearch/k-NN \
+ \
+    && git checkout v1.4.0.0 \
+ \
+    && mkdir /tmp/jni \
+ \
+    && cp jni/src/v1736/* /tmp/jni \
+ \
+    && cd /tmp/jni \
+ \
+    && g++ -fPIC -I/opt/jdk-12.0.2/include -I/opt/jdk-12.0.2/include/linux -I/usr/share/elasticsearch/nmslib/similarity_search/include -std=c++11 -shared -o libKNNIndexV1_7_3_6.so com_amazon_opendistroforelasticsearch_knn_index_v1736_KNNIndex.cpp -lNonMetricSpaceLib -L/usr/share/elasticsearch/nmslib/similarity_search/release \
+ \
+    && mv /tmp/jni/libKNNIndexV1_7_3_6.so /usr/share/elasticsearch/k-NN/buildSrc \
+ \
+    && cd /usr/share/elasticsearch/k-NN \
+ \
+    && ./gradlew build -x :integTestRunner -x :dumpCoverage -x :jacocoTestReport -x test \
```

Comment:
We do not have to run this step. we can remove it.

Reply:
Removed.

Reply:
After removing this step the final docker image size is almost same as before.